Source/VisualRecognitionV3/Models/Class.swift

```diff
@@ -0,0 +1,54 @@
+/**
+ * Copyright IBM Corporation 2018
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ **/
+
+import Foundation
+
+/** A category within a classifier. */
+public struct Class {
+
+    /// The name of the class.
+    public var className: String
+
+    /**
+     Initialize a `Class` with member variables.
+
+     - parameter className: The name of the class.
+
+     - returns: An initialized `Class`.
+    */
+    public init(className: String) {
+        self.className = className
+    }
+}
+
+extension Class: Codable {
```

Comment:
Another thing I forgot to mention from Chris Bailey's talk about "Codable".  He claimed that the compiler will generate the CodingKeys, encode, and decode functions for you -- all you need to do is claim that the class conforms to "Codable".  I'm sure we'll still have to generate the special code for handling dynamic models, but for static models like this one, we might be able to drop the extension and just add the "Codable" protocol to the class definition.

Reply:
Yup! Just conforming to `Codable`, `Encodable`, or `Decodable` will generate quite a lot by default. We can selectively override some of the generated code with our own implementations, when necessary.

- `CodingKeys`: We almost always need to override the default coding keys because we change naming conventions between Swagger and Swift. So we need to map `parameter_name` to `parameterName`.
- `init(from decoder: Decoder)` and `encode(to encoder: Encoder)`: It turns out that even generic `metadata` properties are encoded/decoded using the regular calls. So like you said, we only really need to generate these whenever there are additional properties. I'll open an issue to work on that.